client/src/test/scala/org/http4s/client/scaffold/package.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2014 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.client
+
+import cats.effect.Async
+import cats.implicits._
+import io.netty.channel.Channel
+import io.netty.channel.ChannelFuture
+
+package object scaffold {
+
+  implicit class NettyChannelFutureSyntax[F[_]](private val fcf: F[ChannelFuture]) extends AnyVal {
+    def liftToFWithChannel(implicit F: Async[F]): F[Channel] =
+      fcf.flatMap(cf =>
+        F.async { (callback: Either[Throwable, Channel] => Unit) =>
+          cf.addListener { (f: ChannelFuture) =>
+            if (f.isSuccess) callback(Right(f.channel()))
+            else callback(Left(f.cause()))
+          }
+          ()
+        }
+      )
+  }
+
+  implicit class NettyFutureSyntax[F[_], A <: io.netty.util.concurrent.Future[_]](
```

Comment:
In the `0.23` one it has manual cancellation of `Future` `https://github.com/http4s/http4s/pull/5587/files#diff-23445af51d890997454c978e3ab1c2e9c7ae319fa2b49873807b311a8466c009R55`